src/state/project-state.ts

```diff
@@ -1,53 +1,53 @@
-namespace App {
-  type Listener<T> = (items: T[]) => void;
-  class State<T> {
-    protected listeners: Listener<T>[] = [];
-    addListener(listenerFn: Listener<T>) {
-      this.listeners.push(listenerFn);
-    }
+import { Project } from "../models/project.js";
+import { ProjectStatus } from "../models/project.js";
+type Listener<T> = (items: T[]) => void;
+class State<T> {
+  protected listeners: Listener<T>[] = [];
+  //array of listeners where we store the functions that need to be called whenever the state changes or we add a project
+  addListener(listenerFn: Listener<T>) {
+    this.listeners.push(listenerFn);
+  }
+}
+export class ProjectState extends State<Project> {
+  private projects: Project[] = [];
+  //singleton we can only have one instance of project state class
+  private static instance: ProjectState;
+  private constructor() {
+    super();
   }
-  export class ProjectState extends State<Project> {
-    private projects: Project[] = [];
-    private static instance: ProjectState;
-    private constructor() {
-      super();
-    }
 
-    static getInstance() {
-      if (this.instance) {
-        return this.instance;
-      }
-      this.instance = new ProjectState();
+  static getInstance() {
+    if (this.instance) {
       return this.instance;
     }
-    public addProject(
-      title: string,
-      description: string,
-      people: number
-    ): void {
-      const newProject = new Project(
-        Math.random().toString(),
-        title,
-        description,
-        people,
-        ProjectStatus.Active
-      );
-      this.projects.push(newProject);
+    this.instance = new ProjectState();
+    return this.instance;
+  }
+  //when creting a project we add a random id and set it to do
```

Comment:
Maybe we need a default value that is equal to TODO state and use the default value here